### ISSUE TITLE
[bug] basedpyright exits with status code 1 if there are any warnings and `--warnings` isn't passed

### ISSUE THREAD

**Issue by iamthebot** [DetachHead/basedpyright#1319]
### Description

In theory, based on the CLI help if `--warnings` is *not* passed then basedpyright should return exit code 0 if there are no errors but there are warnings.

But instead I'm getting exit code 1 in that case.

For example, a codebase with a single file like:

```python
#foo.py
from typing import Dict


def foobar(d: Dict[str, str]) -> Dict[str, str]:
    return d
```

Returns diagnostics like:

```
> basedpyright

... warning: This type is deprecated as of Python 3.9; use "dict" instead (reportDeprecated)
... warning: This type is deprecated as of Python 3.9; use "dict" instead (reportDeprecated)
... warning: This type is deprecated as of Python 3.9; use "dict" instead (reportDeprecated)
0 errors, 3 warnings, 0 notes
```

But the exit code is `1` not `0`. Passing `--warnings` produces the same exit code, 1.

I'd think the desired behavior is exit code 0 unless `--warnings` is passed.

This is tested with:

```
basedpyright 1.29.2
based on pyright 1.1.401
```

**Comment by DetachHead** [comment]
this is due to the `failOnWarnings` setting, see https://docs.basedpyright.com/latest/benefits-over-pyright/better-defaults/#typecheckingmode

you can disable it by setting `failOnWarnings` to `false` or setting `typeCheckingMode` to something less strict (eg. `"strict"`, `"standard"`, `"basic"` or `"off"`)

more info about the motivation for basedpyright's strictness in https://github.com/DetachHead/basedpyright/issues/603#issuecomment-2303297625

**Comment by iamthebot** [comment]
Ah, that makes sense. Closing this.
